astor/code_gen.py

```diff
@@ -226,6 +226,32 @@ def body_or_else(self, node):
         self.body(node.body)
         self.else_body(node.orelse)
 
+    def docstring(self, node, indent=1):
+        """
+        Write the function/module/class docstring attribute
+
+        This method has an effect only for python >= 3.7 . For python earlier
+        versions the docstring was represented as the first expression in the
+        function/module/class body.
+
+        Args:
+            node (ast.AST): The node whose docstring is printed.
+            indent (int): How much to indent the docstring (usually is 0 for
+                ast.Module, and 1 for ast.FunctionDef, ast.AsyncFunctionDef and
+                ast.ClassDef).
+        """
+        if not sys.version_info >= (3, 7):
+            return
+
+        value = getattr(node, 'docstring', None)
+        if value is None:
+            return
+
+        self.newline()
+        self.indentation += indent
+        self.write('"""%s"""' % _prep_triple_quotes(value))
```

Comment:
This is just an idea, what do you think about making the following pattern a public function?

```py
'"""%s"""' % _prep_triple_quotes(value)
```

This could be useful for third-party software and we could even reuse it in `pretty_string()`:

https://github.com/berkerpeksag/astor/blob/3c4efdb936cb98892d1915d8d4463b5ba772b935/astor/string_repr.py#L94

Reply:
That sounds useful. What should be the function's name?

Reply:
By public - did you mean to also export it in `__init__.py`?

Reply:
> By public - did you mean to also export it in __init__.py?

Yes and document it in docs/index.rst. By the way, we can deal with this function in another PR and keep this PR focused on adding Python 3.7 support.

I'll have a look at the rest later today, thanks!